Adds OpenProps + PostCSS template

joshnuss opened the pull request:
- Adds a template for OpenProps
- Adds icon for OpenProps
- Removed `autoprefixer` package and replaced `stage` in `postcss-preset-env` config with `minimumVendorImplementations` in PostCSS config
  Based on suggestion here: https://twitter.com/romainmenke/status/1682516646462648320 

## Screenshot

![image](https://github.com/SvelteLab/SvelteLab/assets/4437/0470857d-1a04-4e35-910e-f5a12be34eb9)

joshnuss commented:
**Note**: Casing of template anem is a bit off

 `Openprops` should be `OpenProps`, and `Postcss` should be `PostCSS`.
 
 But I think it's a different problem, ie. templates should have a way to specify it's label.

joshnuss commented:
Should we call the template "OpenProps" or "OpenProps + PostCSS"?

I see other templates are using PostCSS too, so maybe specifying that isn't necessary?

SarcevicAntonio commented:
I'll try my best to take a look ASAP.
Probably tomorrow evening, latest Monday evening GMT+2 ✌️

SarcevicAntonio commented:
OH and Thanks so much for the PR 😁

joshnuss commented:
Most welcome!

And no rush, enjoy the weekend 🌞

paoloricciuti commented on src/lib/default_project_files/openprops/package-lock.json:
We usually delete the package-lock to avoid sending too much over the wire.

paoloricciuti commented:
Other than the only comment I'd say it's ok...as per the issue of the naming I was thinking: could you create a map of "overrides" and look there leaving the fallback to the folder name transformation? I did this because I wanted to avoid having two places to modify when adding a new template but if we have a fallback should be less of a hassle

paoloricciuti commented:
Ah also would you consider handling dark mode in global.css (it just need a media prefers, don't worry about the saved preference of the user)

joshnuss commented:
Thanks @paoloricciuti!

- Removed the `package-lock.json`
- Added dark mode styling. Let me know if that looks OK, (I didn't bring over the `--sk-` vars from the basic template, figured OpenProps provides a similar thing already)

paoloricciuti commented:
> Thanks @paoloricciuti!
> 
> * Removed the `package-lock.json`
> * Added dark mode styling. Let me know if that looks OK, (I didn't bring over the `--sk-` vars from the basic template, figured OpenProps provides a similar thing already)

Thanks to you for contributing :pray:

LGTM!

joshnuss commented:
> per the issue of the naming..

Could also be solved with translations. Like if SvelteLab gets translated into other languages, then the translation for "openprops" could correct the case.

paoloricciuti commented:
> > per the issue of the naming..
> 
> Could also be solved with translations. Like if SvelteLab gets translated into other languages, then the translation for "openprops" could correct the case.

I don't think we plan to translate sveltelab...so I'll probably fix this with the overrides as soon as I have time. Thanks again for contributing 🎉

joshnuss commented:
Welcome!

> I don't think we plan to translate sveltelab...so I'll probably fix this with the overrides as soon as I have time.
Sounds good 👍🏻
